cpp/src/command_classes/UserCode.cpp

```diff
@@ -264,7 +264,8 @@ bool UserCode::HandleMsg
 				memcpy( data, &_data[3], size );
 				value->OnValueRefreshed( data, size );
 			} else {
-				value->OnValueRefreshed( 0x00, 1);
+				size = 0;
+				value->OnValueRefreshed( (uint8 const*)&size, 1);
 			}
```

Comment:
I'm not familiar enough with the coding styles of the openzwave maintainers, but personally, I would be inclined to code this using something like the following (I try to avoid casts if I can).
```
uint8 zero = 0;
value->OnValueRefreshed(&zero, 1);
```
but otherwise, your PR looks good. It might be worth mentioning that the bug being fixed was introduced by commit 0d716a3526eea2c23691c90073c3c5895af3bd24